### PR TITLE
feat!: use results on fs module

### LINKED ISSUE #169: Update Fs module to use Results

### PR DESCRIPTION
**Pull request by gnkz**
Refactor the fs module to use results and errors.

Closes #169

### CODE REVIEW

File: src/_modules/Fs.sol [hunk 1]
```diff
@@ -14,125 +20,339 @@ struct FsMetadata {
...
+        } catch (bytes memory reason) {
+            return FsErrors.FailedToGetCode(abi.decode(removeSelector(reason), (string))).toBytesResult();
+        }
     }
 }
+
+library FsErrors {
+    using LibError for *;
+
+    function FailedToRead(string memory reason) internal pure returns (Error) {
```
Comment (vdrg) [inline src/_modules/Fs.sol:245]: Should these encode the reason in the error message?

**Review by vdrg — CHANGES_REQUESTED** [review]

File: src/_modules/Fs.sol [hunk 2]
```diff
@@ -14,125 +19,335 @@ struct FsMetadata {
...
+
+    function FailedToRemoveFile(string memory reason) internal pure returns (Error) {
+        return FailedToRemoveFile.encodeError("Failed to remove file", reason);
+    }
+
+    function FailedToGetCode(string memory reason) internal pure returns (Error) {
+        return FailedToGetCode.encodeError("Failed to get code", reason);
+    }
+
+    function toStringResult(Error self) internal pure returns (StringResult) {
```
[the remark this answers is not in the document]
Comment (gnkz) [reply src/_modules/Fs.sol]: Done!

File: src/_modules/Fs.sol (hunk 1 above)
Comment (gnkz) [reply src/_modules/Fs.sol:246]: Done!

**Review by vdrg — APPROVED** [review]
lgtm!

**Pull request merged**